src/modeler/AutoRest.Swagger/Validation/AvoidNestedProperties.cs

```diff
@@ -35,8 +35,7 @@ public override bool IsValid(Schema schema, RuleContext context)
 
         private static bool IsClientFlattenUsed(Dictionary<string, object> extensions)
             => extensions.ContainsKey(ClientFlattenExtensionName)
-            && extensions[ClientFlattenExtensionName] is bool
-            && (bool)extensions[ClientFlattenExtensionName] == true;
+            && extensions[ClientFlattenExtensionName] is bool;
```

Comment:
could you add a valid test example for this rule with explicit x-ms-client-flatten set to false? could you also add a comment somewhere here, to indicate that not checking on the value is on purpose?